src/components/nve-switch/nve-switch.component.ts

```diff
@@ -0,0 +1,148 @@
+import { customElement, property, query, state } from 'lit/decorators.js';
+import { INveComponent } from '@interfaces/NveComponent.interface';
+import { CSSResultArray, html, LitElement, TemplateResult } from 'lit';
+import { watch } from '../../utils/watch';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { live } from 'lit/directives/live.js';
+import { classMap } from 'lit/directives/class-map.js';
+import styles from './nve-switch.styles';
+/**
+ * En toggle-button med støtte for ikoner.
+ *
+ * @slot label - Tekst etter switch
+ * @slot onicon - Det første ikonet (til venstre). Synlig når status er PÅ
+ * @slot officon - Det andre ikonet (til høyre). Synlig når status er AV
+ *
+ * @event blur
+ * @event change
+ * @event input
+ * @event focus
+ */
+@customElement('nve-switch')
+export default class NveSwitch extends LitElement implements INveComponent {
+  constructor() {
+    super();
+  }
+  @property({ reflect: true, type: String }) testId: string = '';
+
+  /** Hidden checkbox som holder state */
+  @query('input[type="checkbox"]') input!: HTMLInputElement;
+  @state() private hasFocus = false;
+  @property() title = ''; // make reactive to pass through
+
+  /** Navn på switch */
+  @property() name = '';
+
+  /** Switchens verdi, bruk i forms */
+  @property() value: string = '';
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /** Verdien til switchen. */
+  @property({ type: Boolean, reflect: true }) checked = false;
+  static styles: CSSResultArray = [styles];
+  private emit(eventname: string): void {
+    const event = new CustomEvent(eventname, {
+      bubbles: true,
+      cancelable: false,
+      composed: true,
+      detail: {},
+    });
+    this.dispatchEvent(event);
+  }
+
+  private handleBlur() {
+    this.hasFocus = false;
+    this.emit('blur');
+  }
+
+  private handleInput() {
+    this.emit('input');
+  }
+
+  private handleClick() {
+    this.checked = !this.checked;
+    this.emit('change');
```

Comment:
jeg tror vi skal emite nve- eventer. Shoelace komponenter emitter sl- eventer, vi snakket litt om det, men ble ikke helt enige om hvordan vi skal håndtere det, men @gruble  sa at vi burde kanskje bruke nve- eventer. Gjerne svar på denne kommentaren @gruble.

Reply:
Hmm. Shoelace hadde et argument for å prefikse alle sine hendelser med 'sl-': 
_All custom events are prefixed with sl- to prevent collisions with standard events and other libraries. Refer to a component’s documentation for a complete list of its custom events._ (https://shoelace.style/getting-started/usage#events).
Hvis vi mener dette er et godt argument for å prefikse våre hendelser, er jeg for å prefikse våre med 'nve-' .
Men jeg har ikke nok erfaring til å vurdere om dette er smart eller ikke. Dette må dere frontend-eksperter bli enige om.

Reply:
Jeg er 100% uenig i dette. Dersom det er custom-eventer som ikke finnes fra før (for eksempel en event om at man åpner/lukker en accordion) så kan man bruke det ("nve-toggle"). Men siden vi her lager det som må sier å være en stylet checkbox så føler jeg at den skal bruke samme eventer som en checkbox gjør.

Husk at eventene bobler oppover, så man kan ha
```
<form onchange="setdirty">
<input type="text" name="name" />
<nve-checkbox name="admin />
</form>
```

Dersom vi bruker egne navn, så må vi ha flere event-listeners på denne

Reply:
vet ikke hvilken påvirkning det skal ha når vi skal bruke vanlige eventer her. Jeg syns det allerede er litt spesielt at vi må bruke shoelace eventer og vi skal blande med vanlige eventer. @tomapedersen har du testet det med vanlige eventer: 
`However, it’s important to note that many events emitted within a component’s shadow root will be [retargeted](https://dom.spec.whatwg.org/#retarget) to the host element. This may result in, for example, multiple click handlers executing even if the user clicks just once. Furthermore, event.target will point to the host element, making things even more confusing.`
det kommer fra shoelace docs sida (lenken fra Øystein sin kommentar)

Reply:
De eventene som bubbler fanger jeg ikke. Men ved click og keydown (høyre / venstre piltast) så vil det ikke komme eventer som default. Så da fyrer jeg av "change"-eventer manuelt. Det vil være helt feil dersom piltast fyrer av "nve-change", mens click fyrer av "change" (den bubbler fra checkbox)

Har sjekket det, og det fyres kun av en click, en change, en input, en focus og en blur